### PR TITLE
Force broccoli-typescript-compiler to use newer TypeScript

### PR DESCRIPTION
**Pull request by wagenet**
Resolves the warnings visible here https://github.com/emberjs/ember.js/actions/runs/1715562324.

See also https://github.com/tildeio/broccoli-typescript-compiler/pull/110

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -148,6 +148,9 @@
     "testem-failure-only-reporter": "^0.0.1",
     "typescript": "^4.5.4"
   },
+  "resolutions": {
+    "typescript": "^4.5.4"
+  },
```
Comment (chriskrycho) [inline package.json:153]: Question about this: why not update `broccoli-typescript-compiler` instead?

Comment (wagenet) [reply package.json:153]: We're on the latest version. That said, it would make sense to send a PR to the library.

Comment (wagenet) [reply package.json:153]: Maybe https://github.com/tildeio/broccoli-typescript-compiler/pull/110 ?

Comment (chriskrycho) [reply package.json]: Sorry, yeah, I meant “make the change upstream” but that was totally unclear in my phrasing.

**Comment by krisselden** [comment]
why not update it in broccoli-typescript-compiler?

**Comment by wagenet** [comment]
@krisselden see https://github.com/tildeio/broccoli-typescript-compiler/pull/110

**Comment by wagenet** [comment]
See #19938.

**Pull request closed without merging**